### PR TITLE
Check if monument articles exist and show a red/broken link if not.

### PR DESCRIPTION
**Pull request by bvibber**
Per card 154: https://mingle.corp.wikimedia.org/projects/wlm_android_app/cards/154

### CODE REVIEW

File: assets/www/js/app.js [hunk 1]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+				var pageId;
+				$.each( data.query.pages, function( i, item ) {
+					pageId = i;
+				});
+				if( pageId > 0 ) {
+					// Page exists!
+				} else {
+					// No existy.
+					$( '#monument-detail .monument-article a' )
+						.css( 'color', 'red' )
```
Comment (jdlrobson) [inline assets/www/js/app.js]: use a class instead please

Comment (bvibber) [inline assets/www/js/app.js]: crap i left that in without turning it into a class -- thanks for catching :D

File: assets/www/js/app.js [hunk 2]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+						.click( function( event ) {
```
Comment (jdlrobson) [inline assets/www/js/app.js]: word of caution we use ev and e elsewhere in our code. We should pick one and  be consistent  (my preference is ev)

File: assets/www/messages/messages-en.properties [hunk 3]
```diff
@@ -81,6 +81,7 @@ monument-address=Address:
 monument-country=Campaign:
 monument-no-address=No address found!
 monument-article-link=Read about '$1' on Wikipedia
+article-does-not-exist=Article '$1' does not yet exist on Wikipedia.
```
Comment (jdlrobson) [inline assets/www/messages/messages-en.properties:84]: please add a qqq code to go with this

File: assets/www/js/app.js [hunk 4]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
 		
 		if( monument.articleLink ) {
 			$( '#monument-detail' ).find( '.monument-article' ).html( buildMonumentLink( monument ) );
+			// Run an existence check on the article...
+			var api = new Api( 'https://' + monument.lang + '.wikipedia.org/w/api.php' );
```
Comment (jdlrobson) [inline assets/www/js/app.js]: add to config.js to allow easy changing of this url

File: assets/www/js/app.js [hunk 5]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+			api.request( 'get', {
+				action: 'query',
+				prop: 'revisions',
+				titles: monument.monument_article,
```
Comment (jdlrobson) [inline assets/www/js/app.js]: titles or title ?

File: assets/www/js/app.js [hunk 6]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+				rvlimit: 1
+			} ).done( function( data ) {
+				// data.query.pages
```
Comment (jdlrobson) [inline assets/www/js/app.js]: unnecessary comment

File: assets/www/js/app.js [hunk 7]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+				var pageId;
+				$.each( data.query.pages, function( i, item ) {
```
Comment (jdlrobson) [inline assets/www/js/app.js]: what if we get data back without query.pages ? is this possible? /me no trust api

File: assets/www/js/app.js [hunk 8]
```diff
@@ -178,6 +178,31 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
...
+					pageId = i;
+				});
+				if( pageId > 0 ) {
```
Comment (jdlrobson) [inline assets/www/js/app.js]: minor - whitespace before (

File: assets/www/js/app.js (hunk 5 above)
Comment (bvibber) [inline assets/www/js/app.js]: title -- the api accepts multiples, we just only give it one

Comment (bvibber) [inline assets/www/js/app.js]: *titles

File: assets/www/js/app.js (hunk 7 above)
Comment (bvibber) [inline assets/www/js/app.js]: errorrs may be possible yes. in this case, you'll get an (invisible) JS error and the link will be left blue. not the end of the world...

**Pull request merged**